scripts/gen_ssl_cert.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python2
+#
+# Script creates self signed SSL cert and the key necessary to sign it.
+# This is based off of the code in this article:
+#
+#          the common name with the servers FQDN.
+import argparse
+from OpenSSL import crypto
+from socket import gethostname
+import os
+
+
+def create_key(key_file):
+    """Create SSL key in specified key file."""
+    key = crypto.PKey()
+    key.generate_key(crypto.TYPE_RSA, 4096)
+    open(os.path.join(key_file), "wt").write(
```

Comment:
A context manager, very simply, helps control setup/teardown of resources - in this case covering closing that file handle when you're done with it.

python docs, more focused on _creating_ context managers, which you don't need to do here: https://docs.python.org/2/library/contextlib.html

PEP 343 defining the 'with' context manager: https://www.python.org/dev/peps/pep-0343/

Great stack-overflow post on their use: http://stackoverflow.com/questions/3012488/what-is-the-python-with-statement-designed-for